unit-tests/post-processing/test-post-processing-filters.py

```diff
@@ -0,0 +1,469 @@
+        test_config._output_frames.append(load_from_binary(os.path.join(repo.build, 'unit-tests', 'recordings', test_config.output_frame_names[i])))
+
+    test_config.input_res_x = input_meta_params.input_res_x
+    test_config.input_res_y = input_meta_params.input_res_y
+    test_config.output_res_x = output_meta_params.input_res_x
+    test_config.output_res_y = output_meta_params.input_res_y
+    test_config.depth_units = input_meta_params.depth_units
+    test_config.focal_length = input_meta_params.focal_length
+    test_config.stereo_baseline_mm = input_meta_params.stereo_baseline_mm
+    test_config.downsample_scale = output_meta_params.downsample_scale
+    test_config.spatial_filter = output_meta_params.spatial_filter
+    test_config.spatial_alpha = output_meta_params.spatial_alpha
+    test_config.spatial_delta = output_meta_params.spatial_delta
+    test_config.spatial_iterations = output_meta_params.spatial_iterations
+    test_config.holes_filter = output_meta_params.holes_filter
+    test_config.holes_filling_mode = output_meta_params.holes_filling_mode
+    test_config.temporal_filter = output_meta_params.temporal_filter
+    test_config.temporal_alpha = output_meta_params.temporal_alpha
+    test_config.temporal_delta = output_meta_params.temporal_delta
+    test_config.temporal_persistence = output_meta_params.temporal_persistence
+
+    return check_load_test_configuration(test_config)
+
+def profile_diffs(distances, max_allowed_std, outlier):
+    if distances:
+        mean = sum(distances) / len(distances)
+    else:
+        return
+
+    max_value = max(distances)
+    test.check_equal(max_value, 0)
+
+    e = 0
+    inverse = 1/len(distances)
+    for distance in distances:
+        e += (distance - mean) ** 2
+    standard_deviation = (inverse * e) ** 0.5
+
+    test.check(standard_deviation <= max_allowed_std)
+    test.check(abs(max_value) <= outlier)
+
+    return (abs(max_value) <= outlier) and (standard_deviation <= max_allowed_std)
+
+def validate_ppf_results(result_depth, reference_data, frame_idx):
+    diff2orig = []
+
+    domain_transform_only = (reference_data.downsample_scale == 1) and (not reference_data.spatial_filter) and (not reference_data.temporal_filter)
+    result_profile = result_depth.get_profile().as_video_stream_profile()
+    test.check_equal(result_profile.width(), reference_data.output_res_x)
+    test.check_equal(result_profile.height(), reference_data.output_res_y)
+
+    #Pixel-by-pixel comparison of the resulted filtered depth vs data ercorded with external tool
+    v1 = bytearray(result_depth.get_data())
+    v2 = (reference_data._output_frames[frame_idx])
+    diff2ref = [abs(byte1 - byte2) for byte1, byte2 in zip(v1, v2)]
+
+    #validating depth<->disparity domain transformation is lostless.
+    if domain_transform_only:
+        test.check(profile_diffs(diff2orig, 0, 0))
```

Comment:
Since `diff2orig` is empty, `profile_diffs` will return and we will fail the check. Is this the desired outcome?

Reply:
This line is unnecessary, it does not do anything. I'll remove it.